### PR TITLE
PERF: Styler

### PR DESCRIPTION
**Pull request by jbrockmendel**
```
from asv_bench.benchmarks.io.style import *
self = Render()
self.setup(36, 12)

%timeit self.setup(36, 12); self.time_apply_render(36, 12)
29.2 ms ± 1.2 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)  # <- master
9.98 ms ± 706 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)  # <- PR
```

cc @attack68

### CODE REVIEW

**Comment by attack68** [comment]
nice. good spot. lgtm. tests seem unrelated. there might be one or two other places this pattern might be better too.

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -1023,7 +1023,8 @@ def _update_ctx(self, attrs: DataFrame) -> None:
             )
 
         for cn in attrs.columns:
-            for rn, c in attrs[[cn]].itertuples():
+            ser = attrs[cn]
```
Comment (attack68) [inline pandas/io/formats/style.py:1026]:
```suggestion
            j = self.columns.get_loc(cn)
            ser = attrs[cn]
```
(remove repeated j lookup below)

Comment (jbrockmendel) [reply pandas/io/formats/style.py:1026]: can actually go further and just do `for j, cn in enumerate(attrs.columns):`, but will ask you to do that in #43287

Comment (attack68) [reply pandas/io/formats/style.py:1026]: can't becuase of `subset`, we are cross matching objects:
i.e. 
```
self.columns ~ ["a", "b", "c", "d"]
attrs.columns ~ ["b", "d"]
=> j ~ [1, 3]  # not enumerate
```

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**